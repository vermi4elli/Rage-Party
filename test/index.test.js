const { test, expect, describe, afterAll, afterEach } =
  require('@jest/globals');
//const { fastify } = require('../index');
const { start, close } = require('../index');

start();

describe('server test', () => {
  afterAll(() => {
    //fastify.close();
    close();
  });

  test('responds with success on request /register', async done => {
    const response = await fastify.inject({
      method: 'GET',
      url: '/register'
    });

    expect(response.statusCode).toBe(200);
    expect(response.payload).toBe('{"a":"hello"}');
    done();
  });

  test('responds with success on request /', async done => {
    const response = await fastify.inject({
      method: 'GET',
      url: '/'
    });

    expect(response.statusCode).toBe(200);
    done();
  });

  test('responds with error on request /error', async done => {
    expect((await fastify.inject({
      method: 'GET',
      url: '/error'
    })).body).toMatch('500');
    done();
  });
});
