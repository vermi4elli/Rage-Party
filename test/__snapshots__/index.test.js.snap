// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server test POST returns 404 1`] = `"{\\"message\\":\\"Route POST:/status not found\\",\\"error\\":\\"Not Found\\",\\"statusCode\\":404}"`;
